integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/noarkexchange/receive/InternalQueue.java

```diff
@@ -1,150 +1,123 @@
 import org.springframework.stereotype.Component;
 
 import javax.jms.Session;
-import javax.xml.bind.*;
-import javax.xml.namespace.QName;
-import javax.xml.transform.stream.StreamSource;
-import java.io.*;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
 
 import static no.difi.meldingsutveksling.logging.MessageMarkerFactory.markerFrom;
-import static no.difi.meldingsutveksling.nextmove.logging.ConversationResourceMarkers.markerFrom;
 
 /**
- * The idea behind this queue is to avoid losing messages before they are saved in Noark System.
- *
+ * The idea behind this queue is to avoid loosing messages before they are saved in Noark System.
+ * <p>
  * The way it works is that any exceptions that happens in after a message is put on the queue is re-sent to the JMS listener. If
  * the application is restarted the message is also resent.
- *
+ * <p>
  * The JMS listener has the responsibility is to forward the message to the archive system.
- *
  */
 @Component
+@Slf4j
 public class InternalQueue {
 
-    private static final String EXTERNAL = "external";
     private static final String NOARK = "noark";
     private static final String NEXTMOVE = "nextmove";
-    private static final String PUTMSG = "putmessage";
     private static final String DLQ = "ActiveMQ.DLQ";
 
-    private Logger logger = LoggerFactory.getLogger(InternalQueue.class);
-
-    @Autowired
-    private JmsTemplate jmsTemplate;
-
-    private IntegrajonspunktReceiveImpl integrajonspunktReceive;
-
-    @Autowired
-    private IntegrasjonspunktProperties properties;
-
-    @Autowired
-    private EDUCoreSender eduCoreSender;
-
-    @Autowired
-    private ConversationService conversationService;
-
-    @Autowired
-    private NextMoveSender nextMoveSender;
-
-    private NoarkClient noarkClient;
-
-    private EDUCoreFactory eduCoreFactory;
-
-    private static JAXBContext jaxbContextdomain;
-    private static JAXBContext jaxbContext;
-    private static JAXBContext jaxbContextNextmove;
-
-    private final DocumentConverter documentConverter = new DocumentConverter();
-
-    @Autowired
-    InternalQueue(ObjectProvider<IntegrajonspunktReceiveImpl> integrajonspunktReceive,
+    private final JmsTemplate jmsTemplate;
+    private final IntegrasjonspunktProperties properties;
+    private final ConversationService conversationService;
+    private final NextMoveSender nextMoveSender;
+    private final ObjectMapper objectMapper;
+    private final MessageStatusFactory messageStatusFactory;
+    private final SBDFactory createSBD;
+    private final NextMoveMessageService nextMoveMessageService;
+    private final PutMessageRequestFactory putMessageRequestFactory;
+    private final IntegrajonspunktReceiveImpl integrajonspunktReceive;
+    private final NoarkClient noarkClient;
+    private final DocumentConverter documentConverter;
+    private final UUIDGenerator uuidGenerator;
+
+    InternalQueue(JmsTemplate jmsTemplate,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 13 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS00107)